liota/dccs/iotcc.py

```diff
@@ -390,44 +393,59 @@ def remove_reg_entity_details(self, entity_name, reg_entity_id):
             json.dump(msg, f, sort_keys=True, indent=4, ensure_ascii=False)
         f.close()
 
-    def store_edge_system_info(self, uuid, name, prop_dict):
-        """
-        create (can overwrite) edge system info file of UUID.xml, with format of
-        <attributes>
-        <attribute name=attribute name value=attribute value/>
-        …
-        </attributes>
-        except the first attribute is edge system name, all other attributes may vary
-        """
-
-        log.debug("store_edge_system_info")
-        log.debug('{0}:{1}, prop_list: {2}'.format(uuid, name, prop_dict))
-        root = ET.Element("attributes")
-        # add edge system name as an attribute
-        ET.SubElement(root, "attribute", name="edge system name", value=name)
-        # add edge system properties as attributes
+    def write_entity_json_file(self, prop_dict, attribute_list, uuid, remove):
         if prop_dict is not None:
             for key in prop_dict.iterkeys():
                 value = prop_dict[key]
                 if key == 'entity type' or key == 'name' or key == 'device type':
                     continue
-                ET.SubElement(root, "attribute", name=key, value=value)
-        # add time stamp
-        ET.SubElement(root, "attribute", name="LastSeenTimestamp",
-                      value=strftime("%Y-%m-%dT%H:%M:%S", gmtime()))
+                attribute_list.append({key: value})
+        attribute_list.append({"LastSeenTimestamp": strftime("%Y-%m-%dT%H:%M:%S", gmtime())})
+        log.debug('attribute_list: {0}'.format(attribute_list))
+        msg = {
+            "discovery": {
+                "remove": remove,
+                "attributes": attribute_list
+            }
+        }
+        log.debug('msg: {0}'.format(msg))
+        log.debug("store_entity_json_file dev_file_path:{0}".format(self.dev_file_path))
+        file_path = self.dev_file_path + '/' + uuid + '.json'
+        try:
+            with open(file_path, 'w') as f:
+                json.dump(msg, f, sort_keys=True, indent=4, ensure_ascii=False)
+                log.debug('Initialized ' + file_path)
+            f.close()
+        except IOError, err:
+            log.error('Could not open {0} file '.format(file_path) + err)
 
-        log.debug("store_edge_system_info dev_file_path:{0}".format(self.dev_file_path))
-        file_path = self.dev_file_path + '/' + uuid + '.xml'
-        with open(file_path, "w") as fp:
-            fp.write(self.prettify(root))
```

Comment:
If etree and minidom are not being used anymore, you can remove their import statements.